### ISSUE TITLE
Cached feeds with no timestamp or content are making it into the database

### ISSUE THREAD

**Issue by leonardr** [NYPL-Simplified/server_core#449]
Here are a number of entries in the `cachedfeeds` table:

```
  id   |                                      lane_name                                      | languages | timestamp |  type  |                   facets                   |   pagination    | content | license_pool_id 
-------+-------------------------------------------------------------------------------------+-----------+-----------+--------+--------------------------------------------+-----------------+---------+-----------------
 20314 | Horror                                                                              | eng       |           | groups |                                            |                 |         |                
 45191 | Fantasy                                                                             | eng       |           | page   |                                            |                 |         |                
 45336 | Horror                                                                              | eng       |           | page   |                                            |                 |         |                
 57650 | Books Related to The Secrets of Blueberries, Brothers, Moose & Me by Sara Nickerson | eng       |           | groups |                                            |                 |         |          217398
 57590 | Books Related to The Busy Book Bundle by Trish Kuffner                              | eng       |           | groups |                                            |                 |         |          230687
 46394 | Books Related to Drama Queens in the House by Julie Williams                        | eng       |           | groups |                                            |                 |         |          189035
 49987 | Books Related to A Young Wife by Pam Lewis                                          | eng       |           | groups |                                            |                 |         |          144811
 56711 | Books Related to The Immortalist by Scott Britz                                     | eng       |           | groups |                                            |                 |         |          240760
 52304 | Books Related to Blood and Salt by Kim Liggett                                      | eng       |           | groups |                                            |                 |         |          219834
 56117 | Books Related to 64: Una novela by Sigal Ratner-Arias                               | spa       |           | groups |                                            |                 |         |          148418
 55015 | Books Related to Who Was Woodrow Wilson? by Margaret Frith                          | eng       |           | groups |                                            |                 |         |          220273
 57017 | Fantasy                                                                             | eng       |           | groups |                                            |                 |         |                
 57544 | Books Related to The Pack by Tom Pow                                                | eng       |           | groups |                                            |                 |         |          188955
 37174 | Adult Nonfiction                                                                    | chi       |           | groups |                                            |                 |         |                
 37175 | Adult Nonfiction                                                                    | chi       |           | groups | available=all&collection=main&order=author | after=0&size=50 |         |                
 40493 | Poetry                                                                              | eng       |           | groups |                                            |                 |         |                
 40495 | Poetry                                                                              | eng       |           | groups | available=all&collection=main&order=author | after=0&size=50 |         |                
 44269 | Books Related to Hijacking the Runway by Teri Agins                                 | eng       |           | groups |                                            |                 |         |          159959
```

All of these entries have no `timestamp` and no `content`. They are feeds of all types created both by scripts and the web app. They should not be in this table. I believe the problem is that the feed is retrieved using `fetch()` but the `update()` method (which sets timestamp and content) is never called.

I'm not sure what's causing the underlying issue but we need to stop generating these rows, and stop treating these rows as real cachedfeeds if we do find them.

Suggestions:

1. Look at all the places in core and circulation that call `CachedFeed.fetch()` and see if there's one place where `update()` might not get called.

2. In the get_one_or_create call in CachedFeed.fetch(), try to ignore feeds that don't have a content or timestamp. This might be difficult because I don't think get_one_or_create allows you to put additional restrictions on the initial "does this exist?" query.

3. get_one_or_create might not be a good match for this query in general, because `on_multiple='interchangeable'` leads to useless, duplicate feeds sticking around in the database once they're no longer needed. 

Basically, you might have to improve get_one_or_create, or copy its logic for a CachedFeed-specific lookup method.
